matter_server/server/device_controller.py

```diff
@@ -376,13 +376,16 @@ async def remove_node(self, node_id: int) -> None:
         )
         fabric_index = node.attributes[attribute_path]
 
-        await self.chip_controller.SendCommand(
-            nodeid=node_id,
-            endpoint=0,
-            payload=Clusters.OperationalCredentials.Commands.RemoveFabric(
-                fabricIndex=fabric_index,
-            ),
-        )
+        try:
+            await self.chip_controller.SendCommand(
+                nodeid=node_id,
+                endpoint=0,
+                payload=Clusters.OperationalCredentials.Commands.RemoveFabric(
+                    fabricIndex=fabric_index,
+                ),
+            )
+        except ChipStackError as err:
```

Comment:
I don't understand the PR description. The node is popped from the dict already above the call.

We're only ensuring the event is signaled. But we no longer raise so the caller won't be notified of the failure.

Reply:
Yeah so basically the current behavior is you just can't remove the device on the HA side, it stays there and it doesn't get removed until the matter server is restarted. The current idea with this pull request is to signal HA to remove the device and ignore the fact we couldn't remove the fabric which is not necessarily a bad thing but I guess we could go back to the previous behavior and move the signaling so it happens before we call the sdk.

Reply:
Yes, that sounds good. Always send the signal, and raise on failure.

Reply:
Should we go one step further: keep track that it failed but should be removed, and if the device ever pops up again, we remove it then

Reply:
We could do that but I really don't think that would be necessary as nobody else is doing it, from what I've seen apple and google both try one time to remove their fabric from the device and if they fail that's it, and it could also cause issues if we have a lot of devices that have been removed and we just keep on retrying so we would have to also implement an expiration system for that.

Reply:
I wouldn't keep retrying. But when we get notified of data from a node we haven't seen this run before, we could check if we know it or if we tried deleting it in the past.

Reply:
we will not/never receive any node events if we do not resolve/subscribe it so that is a bit a case of the chicken and the egg.